overlord/snapstate/snapstate.go

```diff
@@ -614,7 +614,7 @@ func InstallPath(st *state.State, si *snap.SideInfo, path, instanceName, channel
 		}
 	}
 
-	channel, err = resolveChannel(st, instanceName, channel, deviceCtx)
+	channel, err = resolveChannel(st, instanceName, "", channel, deviceCtx)
```

Comment:
despite the name this can be used for local refreshes, I don't think this works properly atm but in principle this should probably also be snapst.TrackingChannel.